### PR TITLE
Formatting gunnery skills string in one place to be easily updated.

### PR DESCRIPTION
**Pull request by sagnam**
Support for Legacy RPG Gunnery Skills in MekHQ.  Enforcing at least one of single gunnery skill or RPG gunnery skills.

### CODE REVIEW

File: megamek/src/megamek/common/MULParser.java [hunk 1]
```diff
@@ -1062,23 +1062,82 @@ private void setCrewAttributes(Crew crew, Map<String,String> attributes, Entity
...
+            // get RPG skills
+            int gunneryLVal = gunVal;
+            int gunneryMVal = gunVal;
+            int gunneryBVal = gunVal;
+            if (hasRpgGun) {
+                if ((attributes.containsKey(GUNNERYL)) && (attributes.get(GUNNERYL).length() > 0)) {
+                    try {
+                        gunneryLVal = Integer.parseInt(attributes.get(GUNNERYL));
+                    } catch (NumberFormatException ignored) {
+                        // Handled by the next if test.
```
Comment (sixlettervariables) [inline megamek/src/megamek/common/MULParser.java:1102]: If they have a GUNNERY attribute that is valid, when this falls thru you won't receive any helpful message. You'll just silently use the value from that attribute.

Comment (sagnam) [reply megamek/src/megamek/common/MULParser.java:1102]: Updated

File: megamek/src/megamek/common/util/CrewSkillSummaryUtil.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+    public static String getPilotSkillSummary(String gunnery, String gunneryL, String gunneryM, String gunneryB, String piloting, boolean rpgGunnery) {
+        if (rpgGunnery) {
+            return MessageFormat.format(SKILL_SUMMARY_RPG_PILOT, gunneryL, gunneryM, gunneryB, piloting);
+        } else {
+            return MessageFormat.format(SKILL_SUMMARY_PILOT, gunnery, piloting);
+        }
+    }
+
+    public static String getGunnnerySkillSummary(int gunnery, int gunneryL, int gunneryM, int gunneryB, boolean rpgGunnery) {
```
Comment (Windchild292) [inline megamek/src/megamek/common/util/CrewSkillSummaryUtil.java:33]: This is misspelt, there should only be two n's in gunnery, not three.

File: megamek/i18n/megamek/client/messages.properties [hunk 3]
```diff
@@ -2970,7 +2969,7 @@ WeaponAttackAction.GoldenGoose=Golden Goose
 WeaponAttackAction.GunnerHit=gunner incapacitated
 WeaponAttackAction.GunSkill=gunnery skill
 WeaponAttackAction.GunBSkill=gunnery (B) skill
-WeaponAttackAction.GunESkill=gunnery (E) skill
```
[the remark this answers is not in the document]
Comment (sagnam) [reply megamek/i18n/megamek/client/messages.properties:2973]: In 3ed RPG it was called laser.  In A Time of War for small arms they reference a damage type E for energy.  I think that's why it was labeled aToW and called energy.

Screen Shot 2020-06-15 at 12 24 51 PM
Screen Shot 2020-06-15 at 12 24 57 PM

**Comment by sagnam** [comment]
I have a follow on to this change for computing BV using RPG gunnery skills in the "pilotFactor" rather than the single gunnery skill.  In the case of an all energy unit with a crew that has a good laser gunnery and poor ballistic and poor missile skill the BV is artificially depressed.  This is done by keeping track of the ratio of each weapon type to the total offensive BV and then applying that percentage to the pilotFactor for each gunnery type.

**Comment by sixlettervariables** [comment]
@sagnam another PR would be best, thank you again!

**Pull request merged**